myapp/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,22 @@
 class ApplicationController < ActionController::Base
+  before_action :login_user
+  before_action :re_login
+
+  def login(user)
+    token = User.create_login_token
+    session[:login_token] = token
+    user.update(login_token: User.encrypt_login_token(token))
+  end
+
+  def logout
+    session.delete(:login_token)
+  end
+
+  def login_user
+    User.find_by(login_token: User.encrypt_login_token(session[:login_token]))
```

Comment:
修正前の`@login_user ||=`の形で良いと思います。
```suggestion
    @login_user ||= User.find_by(login_token: User.encrypt_login_token(session[:login_token]))
```
`||=`は、左辺がnilであれば右辺を実行し、返り値を左辺に代入する処理を行います。
この形のメソッドは比較的よく利用します。
login_userのようにDBへのアクセスが必要だが何度も呼び出す必要のあるメソッドは、
`インスタンス変数 ||=`をキャッシュに見立てることで、アクセス回数を減らすことができるからです。

Reply:
DBへのアクセス数が減るよう修正しました。